File: resources/exercises/sort/basic/comb/AlgCombSortEntity.c
//RemoteSort

/* BEGIN TEMPLATE */
void combSort() {
	/* BEGIN SOLUTION */
	int gap = getValueCount();
	int swapped;
	do {
		if (gap>1)
			gap /= 1.3;
		swapped = 0;
		int i;
		for (i=0; i+gap<getValueCount(); i++)
			if (!isSmaller(i,i+gap)) {
				swap(i,i+gap);
				swapped =1;
			}
	} while (gap>1 || swapped);
	/* END SOLUTION */
}
/* END TEMPLATE */


void run(){
	combSort();
}

File: resources/exercises/sort/basic/comb/AlgCombSortEntity.py
# BEGIN SOLUTION
gap = getValueCount()
swapped = True
while gap > 1 or swapped:
	if gap > 1:
		gap = int( gap / 1.3)
	swapped = False
	i = 0
	while i+gap <  getValueCount():
		if not isSmaller(i,i+gap):
			swap(i, i+gap)
			swapped = True
		i += 1
# END SOLUTION
